features/include/pcl/features/flare.h

```diff
@@ -0,0 +1,300 @@
+        }
+
+        /** \brief Get min number of neighbours required for the computation of Z axis.
+        *
+        * \return min number of neighbours required for the computation of Z axis.
+        */
+        inline int
+          getMinNeighboursForNormalAxis () const
+        {
+          return (min_neighbors_for_normal_axis_);
+        }
+
+
+        /** \brief Set min number of neighbours required for the computation of X axis.
+        *
+        * \param[in] min_neighbors_for_tangent_axis min number of neighbours required for the computation of X axis.
+        */
+        inline void
+          setMinNeighboursForTangentAxis (int min_neighbors_for_tangent_axis)
+        {
+          min_neighbors_for_tangent_axis_ = min_neighbors_for_tangent_axis;
+        }
+
+        /** \brief Get min number of neighbours required for the computation of X axis.
+        *
+        * \return min number of neighbours required for the computation of X axis.
+        */
+        inline int
+          getMinNeighboursForTangentAxis () const
+        {
+          return (min_neighbors_for_tangent_axis_);
+        }
+
+
+        /** \brief Provide a pointer to the dataset used for the estimation of X axis.
+        * As the estimation of x axis is negligibly affected by surface downsampling,
+        * this method lets to consider a downsampled version of surface_ in the estimation of x axis.  
+        * This is optional, if this is not set, it will only use the data in the
+        * surface_ cloud to estimate the x axis. 
+        * \param[in] cloud a pointer to a PointCloud 
+        */
+        inline void 
+          setSearchSampledSurface(const PointCloudInConstPtr &cloud)
+        {
+          sampled_surface_ = cloud;
+          fake_sampled_surface_ = false;
+        }
+
+        /** \brief Get a pointer to the sampled_surface_ cloud dataset. */
+        inline PointCloudInConstPtr 
+          getSearchSampledSurface() const
+        {
+          return (sampled_surface_);
+        }
+
+        /** \brief Provide a pointer to the search object linked to sampled_surface.
+        * \param[in] tree a pointer to the spatial search object linked to sampled_surface.
+        */
+        inline void 
+          setSearchMethodForSampledSurface (const KdTreePtr &tree) { sampled_tree_ = tree; }
+
+        /** \brief Get a pointer to the search method used for the extimation of x axis. */
+        inline KdTreePtr 
+          getSearchMethodForSampledSurface () const
+        {
+          return (sampled_tree_);
+        }
+
+        /** \brief Get the signed distances of the highest points from the fitted planes. */
+        inline std::vector<SignedDistanceT> & 
+          getSignedDistancesFromHighestPoints () 
+        {
+          return (signed_distances_from_highest_points_);
+        }
+
+  protected:
+    /** \brief This method should get called before starting the actual computation. */
+    virtual bool
```

Comment:
Hm, ok. We need to check how existing PCL classes deal with the situation when they need additional initialization before computation.